Add full width treeview, refs #6375

jraddaoui opened the pull request:
- Includes Andy Koch's treeview created for WBG without their restrictions
- Removes treeview toggle options
- Adds treeview type setting in global settings form
- Fixes culture problems

jraddaoui committed:
Add full width treeview, refs #6375

Includes Andy Koch's treeview created for WBG without their restrictions

jraddaoui committed:
Cosmetic changes

jraddaoui committed:
Remove treeview toggle options

Not fully implemented and reducing load speed

jraddaoui committed:
Add treeview type setting

- Add migration and fixtures
- Add setting to global settings form
- Use setting to load only the selected treeview

jraddaoui committed:
Fix full width treeview culture problems

Records without title in the current culture were removed from the treeview
in the query. It now gets the source culture title and the current culture title,
giving source culture fallback and showing 'Untitled' (in the current culture)
if none is populated

sevein commented:
Can you try to minimize the number of files that we are adding to vendor/ to what's strictly necessary? e.g. I don't think we are going to need vendor/jstree/libs. I'm not sure about jstreegrid.js, are we using it? Also, we already have a copy of jquery-ui in vendor/jquery-ui.js, can we update that one instead? Do we need their css files, images etc? If not, just avoid them.

sevein commented:
What exactly js/history.js does? Do we need it? It looks like excepting IE9 the history API is available pretty much everywhere. Can we just use interact with it without unnecessary compatibility layers? https://developer.mozilla.org/en-US/docs/Web/API/History | http://caniuse.com/#feat=history

sevein commented on apps/qubit/config/view.yml:
```diff
@@ -25,6 +25,8 @@ default:
     # main:
     # print: { media: print, position: last }
+    /vendor/jquery-ui/css/ui-lightness/jquery-ui-1.10.4.custom.min.css:
+    //netdna.bootstrapcdn.com/font-awesome/4.0.3/css/font-awesome.css:
```
We have a local copy of FA, right?

sevein commented on apps/qubit/modules/informationobject/actions/fullWidthTreeViewAction.class.php:
```diff
@@ -0,0 +1,127 @@
+{
+  public function execute($request)
+  {
+    $start_time = microtime(TRUE);
```
Why are we doing this?

jraddaoui committed:
Fix full width treeview in print view

jraddaoui committed:
Clean-ups and fixes

- Removes unneeded images, css and js files
- Adds only jquery.ui.resizable.js (v.1.8.24)
- Uses fontawesome v3 icons instead of v4
- Uses history API to change URL
- Removes timer

jraddaoui committed:
Remove empty line

jraddaoui committed:
Use minified jquery.ui.resizable JS file

jraddaoui committed:
Fix sidebar search-bar

The #treeview div is mandatory for treeview.js to manage the search-bar

jraddaoui committed:
Fix tooltips

A reminder was needed each time a node is hovered to avoid persistent
tooltips after opening a node with the right arrow

jraddaoui commented:
Merged: https://github.com/artefactual/atom/commit/95185345cf944c1e97c2ab5608a482ed0022bc08
